### ISSUE TITLE
registration_ip, last_login_ip are not added to sql_data_array in Customers class, for customer creation

### ISSUE THREAD

**Issue by torvista** [zencart/zencart#4184]
ZC158
While adding my own fields (bump this https://github.com/zencart/zencart/issues/4103), I find that these two fields are not added to the array and subsequently do not get inserted to the customer record.
I don't get the use of this += operator, which seems to be the culprit:
https://github.com/zencart/zencart/blob/d61dc2b7c7bc0f3fb98c45ec74536deb90d8e13c/includes/classes/Customer.php#L698-L701

**Comment by mc12345678** [comment]
Shouldn't: https://github.com/zencart/zencart/blob/d61dc2b7c7bc0f3fb98c45ec74536deb90d8e13c/includes/classes/Customer.php#L698-L701

Be modified as such?:
````diff
-        $sql_data_array += [
+        $sql_data_array = array_merge($sql_data_array, [
            ['fieldName'=>'registration_ip', 'value'=> $data['ip_address'], 'type'=>'string'],
            ['fieldName'=>'last_login_ip', 'value'=> $data['ip_address'], 'type'=>'string'],
-        ];
+        ]);
````
Or perhaps in use of a variable instead:
````diff
-        $sql_data_array += [
+        $add_elements = [
            ['fieldName'=>'registration_ip', 'value'=> $data['ip_address'], 'type'=>'string'],
            ['fieldName'=>'last_login_ip', 'value'=> $data['ip_address'], 'type'=>'string'],
        ];
+        $sql_data_array = array_merge($sql_data_array, $add_elements);
````

Otherwise array_push could be used, but that would require removing the outer brackets around the new data which makes this a little "cumbersome" in coding.

Alternatively each element "pairing" could be added to the array:
````diff
-        $sql_data_array += [
-            ['fieldName'=>'registration_ip', 'value'=> $data['ip_address'], 'type'=>'string'],
-            ['fieldName'=>'last_login_ip', 'value'=> $data['ip_address'], 'type'=>'string'],
-        ];
+        $sql_data_array[] = 
+            ['fieldName'=>'registration_ip', 'value'=> $data['ip_address'], 'type'=>'string'];
+        $sql_data_array[] = 
+            ['fieldName'=>'last_login_ip', 'value'=> $data['ip_address'], 'type'=>'string'];
````

**Comment by torvista** [comment]
I think the last one is the clearest...and easiest to copy. Copying the current code is what led me to find it didn't work.

**Comment by mc12345678** [comment]
Personally, I prefer the second one for resolution where an array is created/populated with all desired elements and then brought into the existing array. If one wants more they can add more elements inside or heaven forbid can comment out each line that is to be excluded.

I also like, when looking from the context of incorporating the change on an "upgrade", that there appear to be fewer edits and they seem straightforward.

Either way, good catch!

**Comment by drbyte** [comment]
The `+=` syntax was intentional, but I'd forgotten that this array has numeric keys and thus the addition/merge fails because the existing array already has 0 and 1 indices matching the 0 and 1 of the array being added.

I've got other updates pending in that class. I'll look at addressing this soon.

In the meantime, for your own custom fields, just add them yourself as you would otherwise.

**Comment by drbyte** [comment]
Fixed in #4207
